prql-compiler/src/sql/translator.rs

```diff
@@ -1438,4 +1437,40 @@ take 20
 
         Ok(())
     }
+
+    #[test]
+    fn test_strings() -> Result<()> {
+        let query: Query = parse(
+            r###"
+        derive [
+            x = "two households'",
+            y = 'two households"',
+            z = f"a {x} b' {y} c",
+            v = f'a {x} b" {y} c',
+        ]
+        "###,
+        )?;
+
+        assert_display_snapshot!((resolve_and_translate(query)?), @r###"
+        SELECT
+          'two households''' AS x,
```

Comment:
FWIW I'm not sure this is quite right. At least it doesn't compile on BigQuery. (not saying it's on you, tbc) 
<img width="472" alt="image" src="https://user-images.githubusercontent.com/5635139/166979905-1ef4bca5-9e4e-42b8-ac14-b2634ce02ab6.png">